### PR TITLE
Support strided load / store in interpret mode

### PR DESCRIPTION
**Pull request by ayaka14732**
Support strided load / store in interpret mode

The aim of the PR is to support this test: https://github.com/google/jax/blob/bb160cf54ef5f69c2a59a4001a3210fd56a7b286/tests/pallas/ops_test.py#L772-L786

### CODE REVIEW

File: jax/_src/state/discharge.py [hunk 1]
```diff
@@ -247,10 +247,21 @@ def index_array(x, indexers):
       continue
     if indexer is None:
       continue
+
+    if all(isinstance(i, indexing.Slice) for i in indexer.indices):
+      t = []
+      for i in indexer.indices:
+        if isinstance(i, indexing.Slice):
```
Comment (superbobry) [inline jax/_src/state/discharge.py:254]: Can we convert any scalar indexer `i` to a single-element slice and then do `squeeze`?

**Comment by ayaka14732** [comment]
Will make a better fix instead

**Comment by ayaka14732** [comment]
reopened because I realised that the original code still makes sense

**Comment by ayaka14732** [comment]
Superseded by #23117

**Pull request closed without merging**